src/codec.rs

```diff
@@ -78,12 +82,20 @@ impl From<Utf8Error> for ParseError {
 /// Encodes and decodes Language Server Protocol messages.
 #[derive(Clone, Debug)]
 pub struct LanguageServerCodec<T> {
+    message_len: Option<usize>,
     _marker: PhantomData<T>,
 }
 
+impl<T> LanguageServerCodec<T> {
+    pub fn reset(&mut self) {
+        *self = Self::default();
+    }
+}
+
 impl<T> Default for LanguageServerCodec<T> {
     fn default() -> Self {
         LanguageServerCodec {
+            message_len: Option::<usize>::default(),
```

Comment:
I think it would be much simpler to express this directly as `None`.

```suggestion
            message_len: None,
```